### PR TITLE
Add housekeeping jobs

### PR DESCRIPTION
**Pull request by balazsmate**

### CODE REVIEW

**Review by robert-hegedus-hero — APPROVED** [review]

File: repository_analysis.go [hunk 1]
```diff
@@ -2096,12 +2101,312 @@ func (r *analysisRepository) CreateAnalysisResultQueueItem(ctx context.Context,
...
+			log.Error().Err(err).Msg("delete old analysis requests failed")
+			return 0, err
+		}
+		analysisRequestIDs = append(analysisRequestIDs, id)
+	}
+	if len(analysisRequestIDs) == 0 {
+		return 0, err
+	}
+
+	err = r.deleteAnalysisRequestExtraValuesByAnalysisRequestIDs(ctx, analysisRequestIDs)
```
Comment (attilarickert93) [inline repository_analysis.go:2145]: pls delete the related items in a transaction

**Review by attilarickert93 — CHANGES_REQUESTED** [review]

File: repository_analysis.go [hunk 2]
```diff
@@ -2096,12 +2102,381 @@ func (r *analysisRepository) CreateAnalysisResultQueueItem(ctx context.Context,
...
 		log.Error().Err(err).Msg("Failed to create cerberus queue item")
-		return uuid.Nil, nil
+		return uuid.Nil, err
 	}
 
 	return cerberusQueueItem.ID, nil
 }
 
+func (r *analysisRepository) DeleteOldCerberusQueueItems(ctx context.Context, cleanupDays, limit int) (int64, error) {
+	query := fmt.Sprintf(`DELETE FROM %s.sk_cerberus_queue_items WHERE queue_item_id IN
```
Comment (attilarickert93) [inline repository_analysis.go:2112]: only remove successfully sent ones

File: repository_analysis.go [hunk 3]
```diff
@@ -2096,12 +2102,381 @@ func (r *analysisRepository) CreateAnalysisResultQueueItem(ctx context.Context,
...
+func (r *analysisRepository) DeleteOldAnalysisRequests(ctx context.Context, limit int) (int64, error) {
+	tx, err := r.getTransaction()
+	if err != nil {
+		log.Error().Err(err).Msg("delete old analysis requests failed")
+		return 0, err
+	}
+
+	txR := *r
+	txR.db = tx
+	query := fmt.Sprintf(`SELECT id FROM %s.sk_analysis_requests WHERE valid_until_time < timezone('utc', now()) AND is_processed LIMIT $1;`, r.dbSchema)
```
Comment (attilarickert93) [inline repository_analysis.go:2132]: would be nice to keep those which have been used, either if it has been used to sent an order msg to instrument, either if it was paired to with a result

File: skeleton.go [hunk 4]
```diff
@@ -594,10 +604,51 @@ func (s *skeleton) processAnalysisResultBatches(ctx context.Context) {
...
 
-func (s *skeleton) cleanUpCerberusQueueItems(ctx context.Context) {
-	// Todo
-	//-- clean up old stuff
-	//delete from astm.sk_cerberus_queue_items where created_at < now()-interval '14 days';
+const limit = 5000
+
+func (s *skeleton) runCleanupJobs() {
+	s.cleanupCerberusQueueItems()
+	s.cleanupAnalysisResults()
```
Comment (attilarickert93) [inline skeleton.go:611]: unused analysis requests can be deleted, thats fine, but im not sure if results and queue items should be deleted over time at all, without making backup

i'd personally prefer some kind of archive mechanism that would copy the old data to a different archive db before deleting, either manually, either automatically. 
imho manual backup once per (half)year would be enough to keep the db size reasonable, performance shouldn't be affected much due to these tables growing over time

**Review by attilarickert93 — CHANGES_REQUESTED** [review]

**Review by attilarickert93 — APPROVED** [review]

**Pull request merged**